nifi-extension-bundles/nifi-kafka-bundle/nifi-kafka-service-shared/src/test/java/org/apache/nifi/kafka/service/verification/KafkaClusterVerifierTest.java

```diff
@@ -0,0 +1,276 @@
+        final List<ConfigVerificationResult> results = verifier.verifyClusterConnectivity(admin, BOOTSTRAP_SERVERS);
+
+        // Assert
+        assertFalse(results.isEmpty());
+
+        // Should have skipped cluster description
+        final ConfigVerificationResult clusterResult = results.stream()
+                .filter(r -> r.getVerificationStepName().contains("Cluster Description"))
+                .findFirst()
+                .orElseThrow();
+
+        assertEquals(SKIPPED, clusterResult.getOutcome());
+        assertTrue(clusterResult.getExplanation().contains("Cluster description failed"));
+
+        verify(logger).error(eq("Describe Cluster failed"), any(Exception.class));
+    }
+
+    @Test
+    void testVerifyClusterConnectivityWithEmptyNodes() {
+        // Arrange
+        final Collection<Node> nodes = List.of();
+        final KafkaFuture<Collection<Node>> nodesFuture = KafkaFuture.completedFuture(nodes);
+        when(admin.describeCluster(any(DescribeClusterOptions.class))).thenReturn(describeClusterResult);
+        when(describeClusterResult.nodes()).thenReturn(nodesFuture);
+
+        // Act
+        final List<ConfigVerificationResult> results = verifier.verifyClusterConnectivity(admin, BOOTSTRAP_SERVERS);
+
+        // Assert
+        assertTrue(results.size() >= 2);
+
+        // Cluster description result
+        assertEquals(SUCCESSFUL, results.getFirst().getOutcome());
+        assertTrue(results.getFirst().getExplanation().contains("Cluster Nodes Found [0]"));
+
+        // Should have a failure for empty nodes
+        final boolean hasEmptyNodesResult = results.stream()
+                .anyMatch(r -> r.getOutcome() == FAILED && r.getExplanation().contains("No nodes found in cluster"));
+        assertTrue(hasEmptyNodesResult);
+    }
+
+    @Test
+    void testVerifyNodeReachabilityWithTimeout() {
+        // Arrange
+        final Node node = new Node(1, "broker1.example.com", 9092);
+        final Collection<Node> nodes = List.of(node);
+
+        final KafkaFuture<Collection<Node>> nodesFuture = KafkaFuture.completedFuture(nodes);
+        when(admin.describeCluster(any(DescribeClusterOptions.class))).thenReturn(describeClusterResult);
+        when(describeClusterResult.nodes()).thenReturn(nodesFuture);
+
+        socketConnector = (host, port, timeout) -> {
+            throw new TimeoutException("Timeout");
+        };
+
+        // Act
+        final List<ConfigVerificationResult> results = verifier.verifyClusterConnectivity(admin, BOOTSTRAP_SERVERS);
+
+        // Assert - Timeout should indicate failure (node not reachable)
+        final boolean hasFailedNodeResult = results.stream()
+                .anyMatch(r -> r.getVerificationStepName().contains("Node 1")
+                        && r.getVerificationStepName().contains("broker1.example.com")
+                        && r.getOutcome() == FAILED
+                        && r.getExplanation().matches("Connection failed: .*: Timeout"));
```

Comment:
Recommend simplifying this to just `Connection failed` or `Timeout` instead of the full pattern